reasoner/processor/reactive/TransformationStream.java

```diff
@@ -121,7 +98,56 @@ public Stream<OUTPUT, OUTPUT> buffer() {
 
     @Override
     public String toString() {
-        return getClass().getSimpleName() + ":" + operator().getClass().getSimpleName();
+        return getClass().getSimpleName();
+    }
+
+    public static class MapStream<INPUT, OUTPUT> extends TransformationStream<INPUT, OUTPUT> {
```

Comment:
so I didn't delve too deep into this PR but fundamentally this is what we discussed changing and it looks ok to me!